### DISCUSSION TITLE
Gmail users can create multiple accounts with the same email

### DISCUSSION THREAD

**Discussion by sahar2339** [fastapi-users/fastapi-users#1423] [Ideas]
## Describe the bug
FastAPI Users doesn't correctly handle Gmail addresses with dots. Gmail treats addresses with and without dots as the same (e.g., `john.doe@gmail.com` and `johndoe@gmail.com` are identical), but FastAPI Users allows creating separate accounts for these variations. This can lead to security and user management issues.

## To Reproduce
1. Set up a FastAPI Users instance with email authentication.
2. Create a user account with the email `john.doe@gmail.com`.
3. Attempt to create another account with the email `johndoe@gmail.com`.
4. Observe that the second account is created successfully, despite representing the same Gmail address.

## Expected behavior
FastAPI Users should recognize that Gmail addresses with different dot placements are equivalent and prevent the creation of duplicate accounts. When attempting to create an account with a Gmail address that's identical except for dot placement, it should either:
- Reject the new account creation, or
- Treat it as a login attempt for the existing account

## Configuration
Python version: 3.12
FastAPI version: 0.111.0
FastAPI Users version: 13.0.0

## Additional context
- This issue only affects Gmail addresses, as other email providers may treat dots differently.
- Implementing a fix will require careful consideration of existing user accounts and potential database migrations.
- A similar approach might be needed for Gmail's "+" alias feature (e.g., `johndoe+alias@gmail.com`).
- Reference: [Gmail Help: Using dots in Gmail addresses](https://support.google.com/mail/answer/7436150?hl=en)

**Comment by frankie567** [comment]
I've tried on several big services like GitHub, and they do not treat Gmail addresses differently. You can very well register another account by adding a dot inside the address. Same for the "+" alias.

I'm not really favorable to blocking that behavior by default. It's quite similar to websites that want to block temporary email addresses. If you want to prevent multiple Gmail accounts, I suggest you add your own logic by overloading the `create` method:

https://github.com/fastapi-users/fastapi-users/blob/c0c4da9a6caaa51c8b14c1d2439a43485bd2fb6d/fastapi_users/manager.py#L110-L147

**Comment by sahar2339** [comment]
Thank you, it seems fair
